test/integration/isolated/strings.go

```diff
@@ -0,0 +1,24 @@
+package isolated
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+)
+
+// RandomName generates a random name of n length with the provided
+// prefix. If prefix is omitted, the then entire name is random char.
+func RandomName(prefix string, n int) string {
```

Comment:
We already have a similar function in `internal/util/test/name.go` - do you think we could use that instead of adding a new one? If the length has to be limited to 32 chars, I think it's fine to modify that to be limited as well. 

I'm also not sure if it's needed to expose tweaking the length of the generated name as that will lead to having unobvious magic numbers across the codebase (why 3, not 4 characters for RunID, etc.). If we need long and short names, I'd suggest having `RandomName` and `RandomShortName` to limit the explosion of different lengths of names.

Reply:
What's the concern of using random names of different lengths? 🤔 This will allow anyone to just use this in whatever setting they need it.

The random IDs don't really matter all that match but if they are shorter (and their length is not limiting anything) they make it easier to comprehend which resources are from which test etc.)

In any case, we can use an updated (unreleased) version of e2e framework with my fix https://github.com/kubernetes-sigs/e2e-framework/commit/bc43ba6527d096a4f6d8eccdc92941ba9f1ff78e which make the upstream version equivalent to this one. Removing this file then.

134eb00a9

Reply:
> What's the concern of using random names of different lengths? 🤔 This will allow anyone to just use this in whatever setting they need it.

My main concern was the exercise that everyone who wants to use this function needs to do: "What particular length of a name do I want to use? Do I even care?". With dumb `RandomName` and `RandomShortName` we wouldn't have to make this decision every time we use them - the choice would be simpler - do I want it to be long (32) or short (let's say, 4).

Reply:
I get your point. Simplicity is a desired trait but inevitably, a function like this - let's pick `RandomShortName` - will be used in a context where it won't work, because said context will require not 4 but 5 characters. The author of this context will want to keep using it with 5 random charts, not 32 (the only alternative then). Adding a third function will obviously be a suboptimal solution. Changing the implementation to return 5 might be breaking to other callers of this func.